Mathlib/Combinatorics/SimpleGraph/Operations.lean

```diff
@@ -124,31 +125,42 @@ end ReplaceVertex
 
 section AddEdge
 
-/-- Given a vertex pair, add the corresponding edge to the graph's edge set if not present. -/
-def addEdge : SimpleGraph V where
-  Adj v w := G.Adj v w ∨ s ≠ t ∧ (s = v ∧ t = w ∨ s = w ∧ t = v)
-  symm v w := by simp_rw [adj_comm]; (conv_lhs => arg 2; arg 2; rw [or_comm]); exact id
+/-- The graph with a single `s-t` edge. It is empty iff `s = t`. -/
+def edge : SimpleGraph V := fromEdgeSet {s(s, t)}
+
+lemma edge_adj (v w : V) : (edge s t).Adj v w ↔ (v = s ∧ w = t ∨ v = t ∧ w = s) ∧ v ≠ w := by
+  rw [edge, fromEdgeSet_adj, Set.mem_singleton_iff, Sym2.eq_iff]
+
+lemma edge_self_eq_bot : edge s s = ⊥ := by
+  ext a b
+  rw [edge_adj, or_self, bot_adj, iff_false, not_and, and_imp, not_ne_iff]
+  exact fun j k ↦ j.trans k.symm
```

Comment:
I see a lot of logical manipulation. Would something like

```lean
  rw [edge_adj, bot_adj]
  tauto
```

work here?

Reply:
`tauto` doesn't work but `aesop` does.